library/blob.c

```diff
@@ -1625,22 +1646,16 @@ static int eblob_write_prepare_disk_ll(struct eblob_backend *b, struct eblob_key
 		goto err_out_rollback;
 
 	/*
-	 * If no footer is set then commit phase would be skipped and
-	 * so iterator could consider record broken because offset+size
-	 * may be outside of blob. So extend blob manually.
-	 *
-	 * Also we need to extend blob if copy if it was requested, because it
-	 * may be plain_write call that does not call commit and thus following
-	 * copy may try to access area outside of base.
+	 * Allocates space for the record. It should be done because if commit phase will be skipped
+	 * or eblob will be restarted before commit phase, the iterator will consider the record broken
+	 * because offset + size may be outside of blob. So extend blob manually.
 	 */
-	if ((b->cfg.blob_flags & EBLOB_NO_FOOTER) || (copy == EBLOB_COPY_RECORD)) {
-		err = eblob_preallocate(wc->data_fd, wc->ctl_data_offset, wc->total_size);
-		eblob_log(b->cfg.log, EBLOB_LOG_DEBUG, "blob: %s: eblob_preallocate: fd: %d, "
-				"size: %" PRIu64 ", err: %zu\n", eblob_dump_id(key->id),
-				wc->data_fd, wc->ctl_data_offset + wc->total_size, err);
-		if (err != 0)
-			goto err_out_rollback;
-	}
+	err = eblob_preallocate(wc->data_fd, wc->ctl_data_offset, wc->total_size);
```

Comment:
They are broken after restart, aren't they?
posix_fallocate() will perform basically a write with zeroes, and this record will be overwritten immediately after, so it will be double write, can we at least skip it for usual writes which do not set either prepare or commit flag?